src/safeds/data/tabular/transformation/_imputer.py

```diff
@@ -107,18 +108,59 @@ def fit(self, table: Table, column_names: list[str] | None) -> Imputer:
         -------
         fitted_transformer : TableTransformer
             The fitted transformer.
+
+        Raises
+        ------
+        UnknownColumnNameError
+            If column_names contain a column name that is missing in the table
+        ValueError
+            If the table contains 0 rows
+        NonNumericColumnError
+            If the strategy is set to either Mean or Median and the specified columns of the table contain non-numerical data
         """
         if column_names is None:
             column_names = table.column_names
         else:
-            missing_columns = set(column_names) - set(table.column_names)
+            missing_columns = sorted(set(column_names) - set(table.column_names))
             if len(missing_columns) > 0:
-                raise UnknownColumnNameError(list(missing_columns))
+                raise UnknownColumnNameError(missing_columns)
+
+        if table.number_of_rows == 0:
+            raise ValueError("The Imputer cannot be fitted because the table contains 0 rows")
+
+        if (isinstance(self._strategy, Imputer.Strategy.Mean | Imputer.Strategy.Median)) and table.keep_only_columns(
+            column_names,
+        ).remove_columns_with_non_numerical_values().number_of_columns < len(
+            column_names,
+        ):
+            raise NonNumericColumnError(
+                str(
+                    sorted(
+                        set(table.keep_only_columns(column_names).column_names)
+                        - set(
+                            table.keep_only_columns(column_names)
+                            .remove_columns_with_non_numerical_values()
+                            .column_names,
+                        ),
+                    ),
+                ),
+            )
```

Comment:
maybe put the NonNumericColumnError before the ValueError to keep the order of Errors the same as in the other transformations